### ISSUE TITLE
When ingesters are "too busy" rulers should slow down.

### ISSUE THREAD

**Issue by colega** [grafana/mimir#6047]
#### Describe the bug

When ingesters are failing the read requests, rulers should slow down the query rate instead of requesting more aggressively.

#### To Reproduce

Overwhelm ingesters.

See the canceled requests/retries graph: 

![image](https://github.com/grafana/mimir/assets/1511481/c7d2fd72-ba08-41a3-9920-a47d17c82cd7)

#### Expected behavior

Rulers should understand that the system is under pressure and delay the evaluations.

#### Additional Context

Related to https://github.com/grafana/mimir/issues/6000

**Comment by 56quarters** [comment]
This is a good use case for the circuit breaker we just added in the ingester client and I'd ask that any solution here tries to use that instead of working around it.

**Comment by colega** [comment]
I agree that the circuit breaker should be used to release pressure. Also as a feature we should think on recovering the failed evaluations.

**Comment by amurray2306** [comment]
Related to these two incidents 
https://docs.google.com/document/d/1gssUXdYJCLY-PiOe3wINL-pRpXsp7BUre0h07wfMB_A/edit#heading=h.gjdgxs
https://docs.google.com/document/d/1xGKNnGQsgoWXPBSlW9roQ83m32S3mtVQggXyvG51XjU/edit#heading=h.gjdgxs
